src/api/app/models/bs_request.rb

```diff
@@ -31,11 +31,16 @@ class SaveError < APIException
   ]
 
   scope :to_accept, -> { where(state: 'new').where('accept_at < ?', DateTime.now) }
+
   # Scopes for collections
   scope :with_actions, -> { joins(:bs_request_actions).distinct.order(priority: :asc, id: :desc) }
   scope :in_states, ->(states) { where(state: states) }
   scope :with_types, ->(types) { where('bs_request_actions.type in (?)', types) }
+  scope :with_review, -> (review, project) { includes(:reviews).where("reviews.state = 'review' and reviews.by_project = ?", project).references(:reviews) }
+  scope :with_project, ->(project) { with_actions.where("bs_request_actions.source_project = ? AND bs_request_actions.target_project = ?", project, project) }
```

Comment:
Actually this can also be an association:

```
has_many : bs_request_actions, foreign_key: 'source_project', primary_key: 'name'
has_many : bs_requests, through: :bs_request_actions
```

Of course you would also need a second association for target_project.
It's not nice to use strings as keys but currently there is no other way. I'm also investigating to switch to ids but that a different story and not that easy ...